SpotifyApi.cs

```diff
@@ -24,6 +24,21 @@ public void SetAccessToken(string accessToken)
         try
         {
             var response = await client.GetAsync($"https://api.spotify.com/v1/artists/{artistId}");
+
+            if (response.StatusCode is HttpStatusCode.Unauthorized)
+            {
+                throw new SpotifyUnauthorizedException();
+            }
+
+            if (response.StatusCode is HttpStatusCode.TooManyRequests)
+            {
+                // Wait specified seconds in retry-after header or default to 5 minutes.
+                var retryInSeconds = response.Headers.RetryAfter?.Delta?.TotalSeconds ?? 60 * 5;
+                logger.LogWarning($"Too many requests. Retrying in {retryInSeconds} seconds.");
+                await Task.Delay(TimeSpan.FromSeconds(retryInSeconds));
```

Comment:
Will this service run as a server?

Reply:
It is currently just a commandline program exiting once it has looped over all tokens once. So it would need to be scheduled as a cronjob somewhere if we want to use it in the current form.

Reply:
Right. Was just wondering because of this `Task.Delay` and how that will behave in a serverless function.

Reply:
It will not play nicely with serverless :D